### ISSUE TITLE
openbsd issue

### ISSUE THREAD

**Issue by ra1nb0w** [reallyenglish/ansible-role-nsd#31]
Hello,
first of all, thanks a lot for the work!
I encountered some bad things on nsd.conf.j2:

```yaml
{# changed to _chroot_  from _conf_ #}                                                                                                                                                                                                                      
{% if nsd_chroot_enable %}                                                                                                                                                                                                                     
    chroot: "{{ nsd_chroot_dir }}"                                                                                                                                                                                                             
{% endif %}     
```
I imaging that this was a mistype.

```yaml
{# moved zone: into the cycle; every element need zone: before #}                                                                                                                                                                                
{% for k, v in nsd_zones | dictsort %}                                                                                                                                                                                                         
zone:    
```
I moved zone: into the cycle for every element otherwise nsd keeps only the last one.

OpenBSD zeus.cloud.net.mmp.it 6.1 GENERIC.MP#8 amd64
NSD version 4.1.15

thanks and sorry if I didn't send the patch.

**Comment by trombik** [comment]
the latter issue has been fixed in c1ee06d607646850eaf02980ca57146f08480fe1

**Comment by trombik** [comment]
the good news is, branch [support_dict_and_list_in_nsd_conf](https://github.com/reallyenglish/ansible-role-nsd/tree/support_dict_and_list_in_nsd_conf), which is to be released as 2.0.0, should work for you.

the bad news is, the role has been re-written to support arbitrary configurations by getting rid of hard-coded configurations in the template (see #32). you will need to re-write your configuration.

there is a remaining issue: `chroot` test case does not exist, thus, not tested. I am planning to fix #22, but not yet. but it should be possible.

**Comment by ra1nb0w** [comment]
thanks for c1ee06d!

support_dict_and_list_in_nsd_conf = great idea!
in fact when I seen your role I was wondering to my self why you didn't used lists of value :)
and the re-write of configuration it's worth the trouble.

chroot: i don't need it now but should be nice.

I close the issue and I follow #22 #32 

Anyway, thanks for your work!
